src/main/java/org/opensearch/knn/common/launchcontrol/LaunchControlUtils.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.common.launchcontrol;
+
+import org.apache.lucene.search.Query;
+import org.opensearch.knn.index.query.KNNQuery;
+import org.opensearch.knn.index.query.nativelib.NativeEngineKnnVectorQuery;
+
+public final class LaunchControlUtils {
+
+    public static Query getNativeKNNQuery(final KNNQuery knnQuery) {
+        if (KNNLaunchControl.isFeatureEnabled(KNNLaunchControl.KNN_LAUNCH_QUERY_REWRITE_ENABLED_SETTING.getKey())) {
+            return new NativeEngineKnnVectorQuery(knnQuery);
+        }
+        return knnQuery;
+    }
```

Comment:
The responsibility of this class is not aligned with the function which it has. If some class says `LaunchControlUtils` it would be saying as per my opinion is a feature A/B is in launched mode or not. It should not be giving query in the return type.

Reply:
I couldn't think of a better name. suggestions welcome

Reply:
KNNPluginFeatureFlags